src/lib.rs

```diff
@@ -1075,6 +1077,40 @@ impl Device {
     pub fn is_bridge(&self) -> bool {
         self.header == HeaderType::Bridge
     }
+
+    fn print_config_space_bytes(&self, num_bytes: usize) {
+        const MAX_BUFFER_SIZE: usize =
+            100 + (3 * PCIE_CONFIG_SIZE) + (PCIE_CONFIG_SIZE / 16 - 16) * 7 + 16 * 6; // for first 16 lines, addresses are 2 character in hex
+
+        let mut buf = ArrayString::<MAX_BUFFER_SIZE>::new();
+
+        let _ = write!(
+            buf,
+            "\n{:02X}:{:02X}.{:02X} {:?} Vendor:{:02X} Device:{:02X} (rev {})",
+            self.bdf.bus.val(),
+            self.bdf.dev.val(),
+            self.bdf.func.val(),
+            self.device_info.desc,
+            self.vendor_id,
+            self.device_id,
+            self.revision,
+        );
+
+        for i in 0..num_bytes {
+            let value = if cfg!(test) {
+                (0x0 + i % 0xff) as u8
+            } else {
+                self.cfg_read8(i as u16)
+            };
+
+            if i % 16 == 0 {
+                let _ = write!(buf, "\n{:02x}: ", i);
+            }
+            let _ = write!(buf, "{:02x} ", value);
+        }
+        let _ = write!(buf, "\n");
+        log::info!("{}", buf);
```

Comment:
Instead of fabbing up this rather large array string on the stack, does it make sense to issue separate log::info!() calls for separate lines?

Reply:
Or just buffer up a single line if you are worried about everything being put on a new line.

Reply:
I was wondering if there is way not to print the log headers (like "[2024-08-09T18:37:17Z INFO  pci") and this was the way I figured.

Reply:
The log prefix seems to be an attribute of the integrating component, right? We probably might be okay with having that displayed. This is a feature useful for debug purposes specifically. So, having the prefix shouldn't hurt.

Allocation of ~14KiB on the stack seems excessive :D.

Let's see if Tim has other ideas.

Reply:
There is some magic thing you can pass into the log::info!() call to make it bypass the prefix, I think it's `target: ""`

Reply:
I've updated to do log::info! every line